### PR TITLE
fix(cloudflare-worker): hibernate tray leader WebSocket to stop runaway DO duration cost

### PR DESCRIPTION
**Pull request by trieloff**
## Why

The `slicc-tray-hub` Cloudflare account was accruing ~$12/day (and climbing) of **Durable Objects Duration** charges ($12.50 per 1M GB-seconds). Analytics for the `SessionTrayDurableObject` namespace showed the smoking gun:

| date | active (s/day) | CPU (s/day) | CPU% |
|---|---|---|---|
| 06-03 | 8,060,854 | 127 | **0.00%** |
| 06-04 | 7,232,696 | 103 | 0.00% |

~7-8M active (in-memory) seconds/day at **~0% CPU** = ~80 DO instances pinned in memory 24/7. Cause: the leader control socket used `server.accept()` + `addEventListener`, which keeps the object resident in memory for the entire WebSocket lifetime, so every connected desktop session bills duration continuously even while idle.

## What

- Switch the leader socket to the **WebSocket Hibernation API**: `state.acceptWebSocket(server, ['leader'])` plus `webSocketMessage` / `webSocketClose` / `webSocketError` handlers, so the runtime evicts the object between messages and stops billing idle time.
- Add `restoreLeaderSocket()` (via `getWebSockets('leader')`) so cross-invocation paths — webhook forwarding, follower signaling, liveness checks — recover the socket after eviction instead of relying on an in-memory field.
- `webSocketClose`/`Error` only `await loadTray()` when the tray isn't already resident, preserving synchronous disconnect marking (and the reclaim-TTL semantics).

Since CPU is ~0%, this should cut that namespace's duration bill by roughly 99%. No migration or `wrangler.jsonc` change is needed (hibernation is an API-level switch on the existing SQLite-backed class).

## Tests

- **Unit:** made the fake DO state hibernation-aware (`acceptWebSocket`/`getWebSockets`, routes messages to the DO handlers) and added a test proving leader-socket recovery after a simulated eviction. Full suite: 193 passing.
- **Post-deploy (`deployed.test.ts`), targeting these edge-runtime edge cases:**
  - webhook forwarded to a hibernatable leader socket on a **separate** HTTP invocation (the exact rehydration path the fix changed — previously unexercised live; the old webhook checks only hit `WEBHOOK_ID_REQUIRED`).
  - webhook delivery after a ~10s idle gap that may trigger hibernation (correct whether or not eviction happens).
  - leader liveness drops after the socket closes so webhooks are rejected with `NO_LIVE_LEADER`.

## Verification

`tsc -p tsconfig.worker.json` clean; `biome check` clean; `vitest --project cloudflare-worker` 193 passing (new live tests skip without `WORKER_BASE_URL`).

### CODE REVIEW

File: packages/cloudflare-worker/src/session-tray.ts [hunk 1]
```diff
@@ -171,6 +168,41 @@ export class SessionTrayDurableObject {
...
+    this.leaderSocket = ws;
+    if (!this.tray) {
+      await this.loadTray();
+    }
+    const data = typeof message === 'string' ? message : new TextDecoder().decode(message);
+    await this.handleLeaderMessage(ws, data);
+  }
+
+  async webSocketClose(ws: TrayWebSocketLike): Promise<void> {
+    this.leaderSocket = ws;
```
[the remark this answers is not in the document]
Comment (trieloff) [reply packages/cloudflare-worker/src/session-tray.ts]: Good catch — fixed in 0e189c76. The close/error handlers now use `getWebSockets('leader')` as the source of truth and skip teardown when another leader socket is still live, so a stale/late close during a reconnect race no longer marks the tray disconnected. Added a unit regression test (`ignores a stale leader close when a newer leader socket is already live`).
